src/FixedRateIrm.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {IIrm} from "../lib/morpho-blue/src/interfaces/IIrm.sol";
+import {IFixedRateIrm} from "./interfaces/IFixedRateIrm.sol";
+
+import {ErrorsLib} from "./libraries/ErrorsLib.sol";
+import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+import {Id, MarketParams, Market} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+/// @title FixedRateIrm
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+contract FixedRateIrm is IFixedRateIrm {
+    using MarketParamsLib for MarketParams;
+
+    /* EVENTS */
+
+    /// @notice Emitted when a borrow rate is set.
+    event SetBorrowRate(Id indexed id, uint256 newBorrowRate);
+
+    /* STORAGE */
+
+    /// @notice Borrow rates.
+    mapping(Id => uint256) public _borrowRate;
+
+    /* SETTER */
+
+    /// @inheritdoc IFixedRateIrm
+    function setBorrowRate(Id id, uint256 newBorrowRate) external {
```

Comment:
Due to lack of access control "market rate squatting" is possible. The only malleable parameter is the oracle so a market creator will need to deploy a custom oracle to bypass a wrongly set rate.

Reply:
>"market rate squatting"

what do you mean?

Reply:
I agree that it seems very annoying, especially if it leads to a lot of new oracles being deployed for nothing